test/DynamicFees.t.sol

```diff
@@ -146,17 +148,186 @@ contract TestDynamicFees is Test, Deployers, GasSnapshot {
         swapRouter.swap(key, params, testSettings, ZERO_BYTES);
         snapEnd();
 
-        assertEq(_fetchPoolSwapFee(key), 123);
+        assertEq(_fetchPoolLPFee(key), 123);
+    }
+
+    function test_updateDynamicLPFee_100PercentLPFee_AmountIn_NoProtocol() public {
+        assertEq(_fetchPoolLPFee(key), 0);
+
+        dynamicFeesHooks.setFee(1000000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -100, 0, SQRT_RATIO_1_1, 1e18, -1, 1000000);
+
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        assertEq(_fetchPoolLPFee(key), 1000000);
+    }
+
+    function test_updateDynamicLPFee_50PercentLPFee_AmountIn_NoProtocol() public {
+        assertEq(_fetchPoolLPFee(key), 0);
+
+        dynamicFeesHooks.setFee(500000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -100, 49, 79228162514264333632135824623, 1e18, -1, 500000);
+
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        assertEq(_fetchPoolLPFee(key), 500000);
+    }
+
+    function test_updateDynamicLPFee_50PercentLPFee_AmountOut_NoProtocol() public {
```

Comment:
msot of these new tests are named `test_updateDynamicLPFee_` but imo it feels like its really a test of swap as a whole not just of that function - so i think i'd rather `test_swap_`